datafusion/expr/src/udf.rs

```diff
@@ -696,6 +696,33 @@ pub trait ScalarUDFImpl: Debug + DynEq + DynHash + Send + Sync {
         Ok(ExprSimplifyResult::Original(args))
     }
 
+    /// Attempts to convert a literal value to in interval of the corresponding datatype
+    /// of a column expression so that a **preimage** can be computed for
+    /// pruning comparison predicates.
+    ///
+    /// This is used during predicate-pushdown optimization
+    /// (see `datafusion-optimizer-udf_preimage::preimage_in_comparison_for_binary`)
+    ///
+    /// Currently is only implemented by:
+    /// - `date_part(YEAR, expr)`
+    ///
+    /// # Arguments:
+    /// * `lit_value`:  The literal `&ScalarValue` used in comparison
+    /// * `target_type`: The datatype of the column expression inside the function
+    ///
+    /// # Returns
+    ///
+    /// Returns an `Interval` of the appropriate target type if a
+    /// preimage cast is supported for the given function/operator combination;
+    /// otherwise returns `None`.
+    fn preimage(
+        &self,
+        _lit_value: &ScalarValue,
+        _target_type: &DataType,
```

Comment:
Each `ScalarValue` knows its type, so I don't think we need to pass in `target_type`, we could get it by calling `ScalarValue::data_type`

https://docs.rs/datafusion/latest/datafusion/common/enum.ScalarValue.html#method.data_type

Reply:
I also think it might be important to pass in a `info: &SimplifyInfo` here